### PR TITLE
Enable the ability to set log level #238

### LINKED ISSUE #238: Support log-level for controller logs
**@jgwest EDIT:**
Enhancement Request: 
Add support for configurable log levels via [parameter](https://github.com/argoproj-labs/applicationset/blob/48a7b47da0552b9e5a518704e25489ab4a36645c/main.go#L81), eg the ability to set additional log levels beyond just the `--debug` parameter support we have now. For example, as requested, ability to set the log level to only show error or warn.

**Original Text from @dorroddorrod**
Hi,
Is there any way to control the log level for the application-set controller ?

### PR DESCRIPTION
**Pull request by rumstead**
closes #238

### CODE REVIEW

**Comment by jgwest** [comment]
@rumstead  Looks great to me, is it still WIP? Any other changes you are planning, or good to go?

**Comment by rumstead** [comment]
G2G

File: main.go [hunk 1]
```diff
@@ -177,3 +178,14 @@ func main() {
...
 	}
 }
+
+func setLoggingLevel(debug bool, logLevel string) {
+	// the debug flag takes precedence over the loglevel flag
+	if debug {
+		log.SetLevel(log.DebugLevel)
+	} else {
+		level, err := log.ParseLevel(logLevel)
+		setupLog.Error(err, "unable to parse loglevel", "loglevel", logLevel)
```
Comment (jgwest) [inline main.go:188]: Forgot that I had one question, @rumstead:

Should this be:
```
if err != nil {
  // log error about unable unable to parse loglevel?
} else {
  set level?
}
```

**Review by jgwest — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (rumstead) [reply main.go]: Yes, good catch! I committed as the wrong user and had a heck of a time undoing them.

**Review by jgwest — APPROVED** [review]
Looks great, thanks again @rumstead!

**Pull request merged**